overlord/snapstate/snapmgr.go

```diff
@@ -178,8 +178,11 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 
 	if ss.Revision.Unset() {
 		// Local revisions start at -1 and go down.
+		// (unless it's a really old local revision in which case it needs fixing)
 		revision := snapst.LocalRevision
-		if revision.Unset() {
+		if revision.Unset() || revision.N > 0 {
+			// XXX: if revision.N>0 this fixes it, but
+			// XXX: rolling back from this is going to be tricky
```

Comment:
so undoing isn't something the code cared about even before (we are just wasting some local revnos), in case the trick would be to set a "old-local-revision" on the task and use it in undo, see similar code for old-channel, old-trymode...